### PR TITLE
more localization for two-factor

### PR DESCRIPTION
**Pull request by gissimo**
I have added more localization strings in template two_factor_setup.html, and updated accordingly ES and IT translations.

### CODE REVIEW

**Comment by jwag956** [comment]
Thanks - and sorry for taking a while to respond. I'd like to try to get the translations for 'primary_method' up into the form/response - rather than in the template itself - we have those strings already localized - just not a reverse look up. And there are 2 places these exist - two-factor and unified signin.
If you have time/inclination to do that - please do - otherwise - let me know and i will work on that - and let you then just add appropriate translations.

**Comment by gissimo** [comment]
Hi! Thanks for feedback. I see your point. I've moved localization to views.py and created there a dictionary (like the one in forms.py) because what is shown now for primary method is a code and not really a string. Also, already existing localization strings are similar ones but not exact ones, so they need to be created anew. How do you feel about it?

File: flask_security/views.py [hunk 1]
```diff
@@ -126,6 +127,15 @@
     webauthn_verify_response,
 )
 
+_tf_methods = {
+    "google_authenticator": _("Google Authenticator"),
+    "authenticator": _("Authenticator app"),
+    "email": _("Email"),
+    "mail": _("Email"),
+    "sms": _("SMS"),
+    None: _("None"),
```
Comment (jwag956) [inline flask_security/views.py:136]: I think you want "None" here - not pythons None!

File: flask_security/views.py [hunk 2]
```diff
@@ -126,6 +127,15 @@
     webauthn_verify_response,
 )
 
+_tf_methods = {
```
Comment (jwag956) [inline flask_security/views.py:130]: I would like to see this in forms.py - that's where all the other labels are - and reference that from views. This should also make it easier to re-use in unified_signin.py which has a very similar localization issue.

**Review by jwag956 — CHANGES_REQUESTED** [review]
Thanks for working on this - a couple small comments.

File: flask_security/views.py (hunk 2 above)
Comment (gissimo) [reply flask_security/views.py:130]: Sure thing, done!

File: flask_security/views.py (hunk 1 above)
Comment (gissimo) [reply flask_security/views.py]: getattr(user, "tf_primary_method", None) in views.py#L884 can return indeed a python None, according to what's stored in user attribute. OK to have both str and None as dictionary keys?

**Review by jwag956 — CHANGES_REQUESTED** [review]
Thanks for the changes.
I think there are a few more places this same translation is needed.
In views::two_factor_token_validation - both render_template set 'chosen_method' - those I think should use the new translation table.

Same in views::two_factor_rescue:render_template

Also - in two_factor_setup - higher up in the code there is a render_template that sets chosen_method for the two_factor_verify_form - this also I think should be translated.

**Comment by gissimo** [comment]
Hi!
I have added translation for _chosen_method_ when it is used as string. However, sometimes it is then used as key inside the template. In these cases, I have left it as it is. To avoid confusion, I have added suffix __xlated_ for when it is a string and translated.
What do you think?

File: flask_security/templates/security/two_factor_setup.html [hunk 3]
```diff
@@ -4,7 +4,7 @@
   On GET:
     choices: Value of SECURITY_TWO_FACTOR_ENABLED_METHODS (with possible addition of 'delete'
     two_factor_required: Value of SECURITY_TWO_FACTOR_REQUIRED
-    primary_method: if a two-factor method has already been set up.
+    primary_method_xlated: if a two-factor method has already been set up.
```
Comment (jwag956) [inline flask_security/templates/security/two_factor_setup.html:7]: While I agree this is confusing - making this change now will break backward compatibility for applications with their own templates. From what I can see - there are no forms where the same value (primary_method) is used both as a key and as an informational piece of info (in other words - from your changes - we never have BOTH primary_method AND primary_method_xlated. So - lets favor keeping things compatible...

File: flask_security/templates/security/two_factor_verify_code.html [hunk 4]
```diff
@@ -4,7 +4,7 @@
 {% block content %}
   {% include "security/_messages.html" %}
   <h1>{{ _fsdomain("Two-factor Authentication") }}</h1>
-  <h2>{{ _fsdomain("Please enter your authentication code generated via: %(method)s", method=chosen_method) }}</h2>
+  <h2>{{ _fsdomain("Please enter your authentication code generated via: %(method)s", method=chosen_method_xlated) }}</h2>
```
Comment (jwag956) [inline flask_security/templates/security/two_factor_verify_code.html:7]: same here - lets keep things compatible.

File: flask_security/views.py [hunk 5]
```diff
@@ -850,7 +851,7 @@ def two_factor_setup():
                 two_factor_setup_form=form,
                 two_factor_verify_code_form=code_form,
                 choices=cv("TWO_FACTOR_ENABLED_METHODS"),
-                chosen_method=pm,
+                chosen_method=pm,  # do not translate
```
Comment (jwag956) [inline flask_security/views.py:854]: like the comment - that will help all of us down the road.

File: flask_security/views.py [hunk 6]
```diff
@@ -880,7 +881,9 @@ def two_factor_setup():
         two_factor_verify_code_form=code_form,
         choices=choices,
         chosen_method=form.setup.data,
-        primary_method=getattr(user, "tf_primary_method", "None"),
+        primary_method_xlated=_tf_methods_xlate[
```
Comment (jwag956) [inline flask_security/views.py:884]: to keep compatibility - lets change all these back to what they were (without the _xlated)

**Review by jwag956 — CHANGES_REQUESTED** [review]
Thanks  for sticking with this. This is a great improvement.
See comments about maintaining backwards compatibility for templaes.

**Comment by gissimo** [comment]
Hi, all change requests applied! Let me know if I can be of any help anymore.

**Review by jwag956 — APPROVED** [review]

**Comment by jwag956** [comment]
Thanks!

**Pull request merged**